### DISCUSSION TITLE
Using monorepo for easier development

### DISCUSSION THREAD

**Discussion by npostulart** [stepci/stepci#34] [Ideas]
Working with the current structure it is not that easy to develop features which affects stepci and runner. Would it be a feasible solution to use a monorepo for development?

**Comment by mishushakov** [comment]
Do you have a suggestion how we could do it? Especially as we're using TypeScript. I know about lerna, but I'd also be interested to explore other options

Thanks for the tip!

**Reply by npostulart** [reply]
I have no direct preference which one to use. I tried Nx, lerna, TurboRepo and simple yarn workspaces before, but did not setup any of them beyond demo use. If I had to choose one of them I would go with lerna first.

**Reply by npostulart** [reply]
If you like I could make up a demo of the monorepo setup using lerna.

**Reply by mishushakov** [reply]
You're welcome! Danke

**Comment by npostulart** [comment]
I added a draft PR which includes a possible way to setup the cli and runner in a monorepo. See https://github.com/stepci/stepci/pull/37 for more.

**Comment by sebastianwis** [comment]
Hi Niklas @npostulart
Thank you once again for your contribution here! 😊
It would be amazing to welcome you to our Discord if you haven't joined it yet:
https://discord.com/invite/KqJJzJ3BTu
This is where we keep users up to date with updates, features and bug resolutions.
